manticore/core/smtlib/solver.py

```diff
@@ -79,6 +63,50 @@ class SolverType(config.ConfigEnum):
 RE_MIN_MAX_OBJECTIVE_EXPR_VALUE = re.compile(r"(?P<expr>.*?)\s+\|->\s+(?P<value>.*)", re.DOTALL)
 
 
+class SolverType(config.ConfigEnum):
+    """Used as configuration constant for choosing solver flavor"""
+
+    z3 = "z3"
+    cvc4 = "cvc4"
+    yices = "yices"
+    auto = "auto"
+    portfolio = "portfolio"
+    boolector = "boolector"
+
+
+consts.add(
+    "solver",
+    default=SolverType.auto,
+    description="Choose default smtlib2 solver (z3, yices, cvc4, boolector, auto)",
+)
+
+
+class SolverInfo:
```

Comment:
Put the config of each solver in each solver class?

Reply:
As Felipe points out, it'd probably be easier to set some abstract defaults on the `Solver` class. You could use `abc.abstractproperty` on `Solver` and override it on `YicesSolver`, `Z3Solver`, etc.

Reply:
I believe we cannot use abstract classes here since we need the command line for each solver and they depend on global options. For instance, `"{consts.z3_bin} -t:{consts.timeout * 1000} -memory:{consts.memory} -smt2 -in` depends on `consts.z3_bin` and `consts.timeout`, and `consts.timeout` which are going to be computed when the command line options are parsed.

Reply:
I followed your advise, but instead of using a property, I had to use a `classmethod`. That was the only way to allow to call `command` and `inits` from the class, without actually create an instance the solvers. Please take a look to my recent changes.